src/Todo/Model.hs

```diff
@@ -0,0 +1,25 @@
+module Todo.Model where
+
+import           Data.List.NonEmpty (NonEmpty (..))
+import qualified Data.Ratio         as R
+import qualified Numeric.Natural    as N
+
+data State = TODO | DONE | FAIL deriving (Eq, Show)
+
+newtype Title = Title (NonEmpty Char) deriving (Eq, Show)
+newtype Priority = Priority (R.Ratio N.Natural) deriving (Eq, Show)
+newtype Tag = Tag (NonEmpty Char) deriving (Eq, Show)
+newtype Deadline = Deadline Int deriving (Eq, Show)
+type Tags = NonEmpty Tag
+
+data Todo = Todo
+    { todoState       :: State
+    , todoTitle       :: Title
+    , todoPriority    :: Priority
+    , todoTags        :: Tags
+    , todoSubtasks    :: [Todo]
```

Comment:
Can be just `Todos`. But if you don't like `Todos` type alias you can safely remove it.